app/controllers/communities_controller.rb

```diff
@@ -4,7 +4,13 @@ class CommunitiesController < ApplicationController
   # GET /communities
   # GET /communities.json
   def index
-    @communities = Community.all
+    if search_params.present?
```

Comment:
Probably could clean this up a bit by doing 

```
@communities = Community.all
@communities = @communities.where(...) if params.present?
```

Possibly breaking the second part into a multiline if, if it gets too long. 

Maybe take it a step further and have a before_action for loading communities all communities on the index. @teknofire ?